### ISSUE TITLE
poetry 1.0.0b5: Key "Extras:" already exists

### ISSUE THREAD

**Issue by stratosgear** [python-poetry/poetry#1585]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.  Well, latest pre-release (1.0.0b5)
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
-  [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: `Linux serenity 5.3.10-arch1-1 #1 SMP PREEMPT Sun, 10 Nov 2019 11:29:38 +0000 x86_64 GNU/Linux`

- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/stratosgear/336ca890ed93be7f7aefd217dc5f6d54

## Issue
I get this:

```
➜ poetry --version
Poetry version 1.0.0b5

➜ poetry install -vvv
Creating virtualenv rapidxmm in /mnt/data/projects/backend/src/.venv
Using virtualenv: /mnt/data/projects/backend/src/.venv

[KeyAlreadyPresent]
Key "extras" already exists.

Traceback (most recent call last):
  File "/home/stratos/.poetry/lib/poetry/_vendor/py3.8/clikit/console_application.py", line 131, in run
    status_code = command.handle(parsed_args, io)
  File "/home/stratos/.poetry/lib/poetry/_vendor/py3.8/clikit/api/command/command.py", line 120, in handle
    status_code = self._do_handle(args, io)
  File "/home/stratos/.poetry/lib/poetry/_vendor/py3.8/clikit/api/command/command.py", line 171, in _do_handle
    return getattr(handler, handler_method)(args, io, self)
  File "/home/stratos/.poetry/lib/poetry/_vendor/py3.8/cleo/commands/command.py", line 92, in wrap_handle
    return self.handle()
  File "/home/stratos/.poetry/lib/poetry/console/commands/install.py", line 63, in handle
    return_code = installer.run()
  File "/home/stratos/.poetry/lib/poetry/installation/installer.py", line 65, in run
    if not self._update and not self._locker.is_locked():
  File "/home/stratos/.poetry/lib/poetry/packages/locker.py", line 48, in is_locked
    return "package" in self.lock_data
  File "/home/stratos/.poetry/lib/poetry/packages/locker.py", line 37, in lock_data
    self._lock_data = self._get_lock_data()
  File "/home/stratos/.poetry/lib/poetry/packages/locker.py", line 220, in _get_lock_data
    return self._lock.read()
  File "/home/stratos/.poetry/lib/poetry/_vendor/py3.8/tomlkit/toml_file.py", line 20, in read
    return loads(f.read())
  File "/home/stratos/.poetry/lib/poetry/_vendor/py3.8/tomlkit/api.py", line 34, in loads
    return parse(string)
  File "/home/stratos/.poetry/lib/poetry/_vendor/py3.8/tomlkit/api.py", line 51, in parse
    return Parser(string).parse()
  File "/home/stratos/.poetry/lib/poetry/_vendor/py3.8/tomlkit/parser.py", line 153, in parse
    key, value = self._parse_table()
  File "/home/stratos/.poetry/lib/poetry/_vendor/py3.8/tomlkit/parser.py", line 1039, in _parse_table
    result = self._parse_aot(result, name)
  File "/home/stratos/.poetry/lib/poetry/_vendor/py3.8/tomlkit/parser.py", line 1083, in _parse_aot
    _, table = self._parse_table(name_first)
  File "/home/stratos/.poetry/lib/poetry/_vendor/py3.8/tomlkit/parser.py", line 1026, in _parse_table
    table.raw_append(key_next, table_next)
  File "/home/stratos/.poetry/lib/poetry/_vendor/py3.8/tomlkit/items.py", line 938, in raw_append
    self._value.append(key, _item)
  File "/home/stratos/.poetry/lib/poetry/_vendor/py3.8/tomlkit/container.py", line 138, in append
    raise KeyAlreadyPresent(key)
```

My poetry settings:

```
➜ poetry config --list
cache-dir = "/home/stratos/.cache/pypoetry"
virtualenvs.create = true
virtualenvs.in-project = true
virtualenvs.path = "."
```

**Comment by stratosgear** [comment]
I bet this has to do with the recent release of Python 3.8, but I have no idea how to fix it... :(

I'm on Arch so I just picked the latest version just today I think

**Comment by sdispater** [comment]
It seems that the error occurs when trying to read the lock file.

What does your lock file look like?

**Comment by stratosgear** [comment]
Like this: https://gist.github.com/stratosgear/c0f7aadd95395609311d502619aaa91e

But thanks for the nudge in the right direction.  Usually when I try to "debug/fix" something like that I remove the `.venv` **and** the `poetry.lock` file, to start clean.  This time I completely messed up and forgot the remove the `poetry.lock`

`poetry install` is working fine now, but that error message was **REALLY** confusing because I was searching for `extras` mention in my `pyproject.toml` (and they were none).

**Comment by sdispater** [comment]
Thanks.

The error comes from the fact that, in the lock file, the extras of `urllib` were specified twice (see https://gist.github.com/stratosgear/c0f7aadd95395609311d502619aaa91e#file-poetry-lock-L1998) which is not valid.

I am not sure how this can happen. Did you recently merge something that could explain the duplicate entries?

That being said, the error message should be clearer I agree.

**Comment by stratosgear** [comment]
I do not remember doing any kind of merging on the `poetry.lock` file.  I rarely have to anyways, I do not remember ever having to do so.  No other developer changed or committed that file.

Since I have already deleted it and recreated it, I have no idea how that duplicate entry was created, sorry.  I'll have my eye on that, next time I get a similar error.

I'll be closing the issue then, since this has been resolved.  Thanks for getting in touch, and keep up the good work.

I really enjoy using poetry (even though I hate the imposed complexity of all that python management and the associated tooling that is required to keep separate unique environments)
